pkg/api/types.go

```diff
@@ -1357,6 +1357,20 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+// DaemonEndpoint store information about a single Daemon endpoint
+type DaemonEndpoint struct {
+	// Port number of the given endpoint
+	Port int `json:port`
+}
+
+// NodeDaemonEndpoints lists ports opened by daemons running on the Node
+type NodeDaemonEndpoints struct {
+	// Endpoint on which Kubelet is listening
+	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty"`
+	// Endpoint opened by Kubelet on which cAdvisor proxy is available
```

Comment:
This may be best answered outside of the context of your PR, but why do we expose cAdvisor on a different port instead of mapping it's functionality under a path on the already exposed kubelet port? It seems like running fewer exposed system level services would be better.